code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilderTest.kt

```diff
@@ -0,0 +1,239 @@
+        mockkObject(PushTemplateImageUtils)
+        mockkObject(BasicNotificationBuilder)
+        mockkObject(PendingIntentUtils)
+        expandedLayout = mockkClass(RemoteViews::class)
+    }
+
+    @Test
+    fun `construct returns NotificationCompat Builder for valid inputs`() {
+        val result = ManualCarouselNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        assertNotNull(result)
+    }
+
+    @Test
+    fun `construct returns BasicNotificationBuilder if download image count is less than 3`() {
+        every { cacheImages(any()) } answers { 2 }
+        ManualCarouselNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        verify {
+            BasicNotificationBuilder.fallbackToBasicNotification(
+                context,
+                trackerActivityClass,
+                broadcastReceiverClass,
+                pushTemplate.data
+            )
+        }
+    }
+
+    @Test
+    fun `construct returns AEPPushNotificationBuilder if download image count is greater than equal to 3`() {
+        every { cacheImages(any()) } answers { 3 }
+        ManualCarouselNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        verify(exactly = 0) {
+            BasicNotificationBuilder.fallbackToBasicNotification(
+                context,
+                trackerActivityClass,
+                broadcastReceiverClass,
+                pushTemplate.data
+            )
+        }
+    }
+
+    @Test
+    fun `test downloadCarouselItems return non empty list for it retrieve the image bitmap from cache successfully`() {
+        every { getCachedImage(any()) } answers { mockkClass(Bitmap::class) }
+        val imagesList =
+            ManualCarouselNotificationBuilder.downloadCarouselItems(pushTemplate.carouselItems)
+        assertFalse(imagesList.isEmpty())
+    }
+
+    @Test
+    fun `test getCarouselIndices with left click intent action`() {
+        pushTemplate.intentAction = PushTemplateConstants.IntentActions.MANUAL_CAROUSEL_LEFT_CLICKED
+        val imageUris = listOf("image1", "image2", "image3")
+        val result = getCarouselIndices(pushTemplate, imageUris)
+        assertEquals(Triple(1, 2, 0), result)
+    }
+
+    @Test
+    fun `test getCarouselIndices with filmstrip left click intent action`() {
+        pushTemplate.intentAction = PushTemplateConstants.IntentActions.FILMSTRIP_LEFT_CLICKED
+        val imageUris = listOf("image1", "image2", "image3")
+        val result = getCarouselIndices(pushTemplate, imageUris)
+        assertEquals(Triple(1, 2, 0), result)
+    }
+
+    @Test
+    fun `test getCarouselIndices with no intent action and filmstrip layout`() {
```

Comment:
Can you add another test case for `test getCarouselIndices with no intent action and manual layout`?

Reply:
done @spoorthipujariadobe